lib/canary/plugs.ex

```diff
@@ -116,6 +116,60 @@ defmodule Canary.Plugs do
   end
 
   @doc """
+  Authorize the current user against the calling controller.
+
+  In order to use this function,
+
+    1) `conn.assigns[Application.get_env(:canary, :current_user, :current_user)]` must be an ecto
+    struct representing the current user
+
+    2) `conn.private` must be a map (this should not be a problem unless you explicitly modified it)
+
+  authorize_controller checks for the name of the current controller in one of the following places
+    1) :phoenix_controller in conn.private
+    2) :canary_controller in conn.assigns
+  In case you are not using phoenix, make sure you set the controller name in the conn.assigns
+  Note that in case neither of :phoenix_controller or :canary_controller are found the requested
+    authorization will automatically fails
```

Comment:
From `controller = conn.assigns[:canary_controller] || conn.private[:phoenix_controller]`, it looks like when neither `phoenix_controller` or `canary_controller` are present, `controller` will be `nil`, meaning that we will just authorize against a `nil` controller in Canada, so authorization will not necessarily fail (this could be set to true in the Canada rules). I think this behaviour is fine since it is consistent with how `authorize_resource` works, but we should make sure the docs here are consistent with the code.